### PR TITLE
[BE] Enable ruff's UP rules and autoformat ao/

### PR DESCRIPTION
**Pull request by justinchuby**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #105437
* #105436
* #105434
* #105433
* #105432
* #105431
* __->__ #105430

### CODE REVIEW

**Review by albanD — APPROVED** [review]
SGTM
FYI @vkuzo

File: torch/ao/quantization/pt2e/quantizer/quantizer.py [hunk 1]
```diff
@@ -128,24 +128,9 @@ class QuantizationConfig:
...
-OperatorConfig = NamedTuple(
-    "OperatorConfig",
-    # fix List[str] with List[List[Union[nn.Module, FunctionType, BuiltinFunctionType]]]
-    # Basically we are mapping a quantization config to some list of patterns.
-    # a pattern is defined as a list of nn module, function or builtin function names
-    # e.g. [nn.Conv2d, torch.relu, torch.add]
-    # We have not resolved whether fusion can be considered internal details of the
-    # quantizer hence it does not need communication to user.
-    # Note this pattern is not really informative since it does not really
-    # tell us the graph structure resulting from the list of ops.
```
Comment (malfet) [inline torch/ao/quantization/pt2e/quantizer/quantizer.py:140]: I would have kept comment in place

**Review by malfet — APPROVED** [review]

File: torch/ao/quantization/pt2e/quantizer/quantizer.py (hunk 1 above)
Comment (justinchuby) [reply torch/ao/quantization/pt2e/quantizer/quantizer.py:140]: Good catch thanks

File: torch/ao/quantization/fx/utils.py [hunk 2]
```diff
@@ -846,12 +843,11 @@ def _activation_post_process_satisfies_dtype_config_constraints(
...
                     not isinstance(activation_post_process, FixedQParamsFakeQuantize):
                 warnings.warn(("QConfig must specify a FixedQParamsObserver or a FixedQParamsFakeQuantize "
-                              "for fixed qparams ops, ignoring %s.\n%s") % (qconfig, suggestion_str))
+                              "for fixed qparams ops, ignoring {}.\n{}").format(qconfig, suggestion_str))
                 return False
             if observer.scale != backend_scale_exact_match or observer.zero_point != backend_zero_point_exact_match:
-                warnings.warn(("QConfig fixed scale (%s) and zero point (%s) do not match the backend's "
-                              "(%s and %s), ignoring %s.\n%s") %
-                              (observer.scale, observer.zero_point, backend_scale_exact_match,
+                warnings.warn(("QConfig fixed scale ({}) and zero point ({}) do not match the backend's "
```
Comment (justinchuby) [inline torch/ao/quantization/fx/utils.py:849]: f-string

File: torch/ao/quantization/observer.py [hunk 3]
```diff
@@ -118,7 +118,7 @@ def _with_callable_args(cls_or_self, **kwargs):
     return r.with_callable_args(**kwargs)
 
 
-ABC: Any = ABCMeta(str("ABC"), (object,), {})  # compatible with Python 2 *and* 3:
+ABC: Any = ABCMeta("ABC", (object,), {})  # compatible with Python 2 *and* 3:
```
Comment (justinchuby) [inline torch/ao/quantization/observer.py:121]: Do we need this for Python 2?

File: torch/ao/quantization/pt2e/quantizer/quantizer.py (hunk 1 above)
Comment (justinchuby) [reply torch/ao/quantization/pt2e/quantizer/quantizer.py:140]: Done

**Pull request closed without merging**